src/rate.rs

```diff
@@ -786,54 +787,34 @@ impl RCState {
           //  count of dropped frames.
         }
         // Increment the frame count for filter adaptation purposes.
-        if self.nframes[fti] < ::std::i32::MAX {
+        if !trial && self.nframes[fti] < ::std::i32::MAX {
           self.nframes[fti] += 1;
         }
       }
-      self.reservoir_fullness += self.bits_per_frame - bits;
-      // If we're too quick filling the buffer and overflow is capped, that
-      //  rate is lost forever.
-      if self.cap_overflow {
-        self.reservoir_fullness =
-          self.reservoir_fullness.min(self.reservoir_max);
-      }
-      // If we're too quick draining the buffer and underflow is capped, don't
-      //  try to make up that rate later.
-      if self.cap_underflow {
-        self.reservoir_fullness = self.reservoir_fullness.max(0);
+      if !trial {
+        self.reservoir_fullness += self.bits_per_frame - bits;
+        // If we're too quick filling the buffer and overflow is capped, that
+        //  rate is lost forever.
+        if self.cap_overflow {
+          self.reservoir_fullness =
+            self.reservoir_fullness.min(self.reservoir_max);
+        }
+        // If we're too quick draining the buffer and underflow is capped, don't
+        //  try to make up that rate later.
+        if self.cap_underflow {
+          self.reservoir_fullness = self.reservoir_fullness.max(0);
+        }
+        // Adjust the bias for the real bits we've used.
+        self.rate_bias +=
+          bexp64(prev_log_scale + self.log_npixels - log_q_exp) - bits;
       }
-      // Adjust the bias for the real bits we've used.
-      self.rate_bias +=
-        bexp64(prev_log_scale + self.log_npixels - log_q_exp) - bits;
     }
     dropped
   }
 
   pub fn needs_trial_encode(&self, fti: usize) -> bool {
       self.target_bitrate > 0 && self.nframes[fti] == 0
   }
-
-  pub fn record_trial_encode(
-    &mut self, bits: i64, fti: usize, log_target_q: i64
-  ) {
-    assert!(self.needs_trial_encode(fti));
-    assert!(bits > 0);
```

Comment:
These two assert!() statements did not make it.  Do we want to still check for them if trial is true?

Reply:
Yes, the above comment is related and where we should confirm the preconditions.